shell-completion/bash/systemd-analyze

```diff
@@ -30,6 +30,15 @@ __get_machines() {
         machinectl list --no-legend --no-pager | { while read a b; do echo " $a"; done; };
 }
 
+__get_syscall_sets() {
+        local line
+        systemd-analyze syscall-filter --no-pager | while IFS= read -r line; do
+                if [[ $line == @* ]]; then
+                        printf '%s\n' "$line"
```

Comment:
I guess that `echo " $line"` is faster.

Reply:
I couldn’t measure any significant difference:

```sh
$ for i in {1..1000000}; do printf '%s\n' "@foo" " bar" " baz"; done > /tmp/output
$ time while IFS= read -r line; do if [[ $line == @* ]]; then printf '%s\n' "$line"; fi; done < /tmp/output > /dev/null

real    1m18,929s
user    1m13,667s
sys     0m5,232s
$ time while IFS= read -r line; do if [[ $line == @* ]]; then echo " $line"; fi; done < /tmp/output > /dev/null

real    1m14,851s
user    1m10,103s
sys     0m4,741s
$ time while IFS= read -r line; do if [[ $line == @* ]]; then printf '%s\n' "$line"; fi; done < /tmp/output > /dev/null

real    1m15,528s
user    1m10,841s
sys     0m4,665s
$ time while IFS= read -r line; do if [[ $line == @* ]]; then echo " $line"; fi; done < /tmp/output > /dev/null

real    1m33,447s
user    1m27,709s
sys     0m5,655s
$ time while IFS= read -r line; do if [[ $line == @* ]]; then printf '%s\n' "$line"; fi; done < /tmp/output > /dev/null

real    1m45,044s
user    1m38,670s
sys     0m6,281s
$ time while IFS= read -r line; do if [[ $line == @* ]]; then echo " $line"; fi; done < /tmp/output > /dev/null

real    1m18,719s
user    1m13,438s
sys     0m5,259s
```

I think the differences here are just due to other load on the system where I was running this.

My reason for preferring `printf` over `echo` is that `echo` is more fragile in general. It’s not as much of a problem here – it’s a bash completion script, and in bash `echo` slightly more sane than in some other shells – but in general, I consider `echo` to be a minor code smell in shell scripts, and use `printf` whenever writing them myself.

Reply:
Yeah, I think `printf` is considered more portable in general.